packages/expo-cli/src/commands/build-native/index.ts

```diff
@@ -70,12 +85,77 @@ export async function buildAction(projectDir: string, options: Options): Promise
   }
 }
 
-async function statusAction(projectDir: string): Promise<void> {
-  throw new Error('not implemented yet');
+async function statusAction(
+  projectDir: string,
+  { platform, status, buildId }: StatusOptions
+): Promise<void> {
+  if (buildId) {
+    if (platform) {
+      throw new Error('-p/--platform cannot be specified if --build-id is specified.');
+    }
+
+    if (status) {
+      throw new Error('-s/--status cannot be specified if --build-id is specified.');
+    }
+  } else {
+    if (platform && !platforms.includes(platform)) {
+      throw new Error(
+        `-p/--platform needs to be one of: ${platforms.map(p => log.chalk.bold(p)).join(', ')}`
+      );
+    }
+
+    if (status && !statuses.includes(status)) {
+      throw new Error(
+        `-s/--status needs to be one of: ${statuses.map(s => log.chalk.bold(s)).join(', ')}`
+      );
+    }
+  }
+
+  const user: User = await UserManager.ensureLoggedInAsync();
+  const { exp }: ProjectConfig = getConfig(projectDir);
+
+  const accountName = exp.owner || user.username;
+  const projectName = exp.slug;
+
+  const projectId = await ensureProjectExistsAsync(user, {
+    accountName,
+    projectName,
+  });
+
+  const client = ApiV2.clientForUser(user);
+
+  const spinner = ora().start('Fetching build history...');
+
+  let builds: BuildInfo[] | undefined;
+
+  try {
+    const params = {
```

Comment:
if it's used only inside else, move this variable there or inline as function argument